storm/src/main/storm/mesos/resources/OfferResources.java

```diff
@@ -0,0 +1,187 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package storm.mesos.resources;
+
+import backtype.storm.scheduler.TopologyDetails;
+import org.apache.mesos.Protos;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import storm.mesos.util.MesosCommon;
+import storm.mesos.util.PrettyProtobuf;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+// TODO(ksoundararaj): a better name? - OfferAggregator?
+public class OfferResources {
+
+  private final Logger log = LoggerFactory.getLogger(OfferResources.class);
+  private Map<ResourceType, Resource> availableResources;
+
+  private List<Protos.Offer> offerList = new ArrayList<Protos.Offer>();
+
+  private final String hostName;
+
+  private Protos.SlaveID slaveID;
+
+  private void initializeAvailableResources() {
+    availableResources = new HashMap<>();
+    availableResources.put(ResourceType.CPU, new ScalarResource(ResourceType.CPU));
+    availableResources.put(ResourceType.MEM, new ScalarResource(ResourceType.MEM));
+    availableResources.put(ResourceType.PORTS, new RangeResource(ResourceType.PORTS));
+  }
+
+  public OfferResources(String hostName) {
+    this.hostName = hostName;
+    initializeAvailableResources();
+  }
+
+  public OfferResources(Protos.Offer offer) {
+    initializeAvailableResources();
+    if (slaveID == null) {
+      this.slaveID = offer.getSlaveId();
+    }
+    this.hostName = offer.getHostname();
+    add(offer);
+  }
+
+  public String getHostName() {
+    return hostName;
+  }
+
+  public void add(Protos.Offer offer) {
+    offerList.add(offer);
+
+    for (Protos.Resource r : offer.getResourcesList()) {
+      if (r.hasReservation()) {
+        // skip resources with dynamic reservations
+        continue;
+      }
+
+      ResourceType resourceType = ResourceType.of(r.getName());
+
+      ReservationType reservationType = (r.getRole().equals("*")) ? ReservationType.UNRESERVED : ReservationType.STATICALLY_RESERVED;
+
+      switch (resourceType) {
+        case CPU:
+        case MEM:
+          ResourceEntries.ScalarResourceEntry scalarResourceEntry = new ResourceEntries.ScalarResourceEntry(reservationType, r.getScalar().getValue());
+          availableResources.get(resourceType).add(scalarResourceEntry, reservationType);
+          break;
+        case PORTS:
+          for (Protos.Value.Range range : r.getRanges().getRangeList()) {
+            ResourceEntries.RangeResourceEntry rangeResourceEntry = new ResourceEntries.RangeResourceEntry(reservationType, range.getBegin(), range.getEnd());
+            availableResources.get(resourceType).add(rangeResourceEntry, reservationType);
+          }
+          break;
+        case DISK:
+          break;
```

Comment:
Maybe plop a TODO into the code there that references #147 and then we can link the ticket to the code.